src/presence/PresenceHeartbeat.js

```diff
@@ -0,0 +1,66 @@
+// @flow strict-local
+import { PureComponent } from 'react';
+import { AppState } from 'react-native';
+import type { Auth, Dispatch } from '../types';
+
+import { connect } from '../react-redux';
+import { tryGetAuth } from '../account/accountsSelectors';
+import { reportPresence } from '../actions';
+import Heartbeat from './heartbeat';
+
+type Props = $ReadOnly<{|
+  dispatch: Dispatch,
+  auth: Auth | void,
+|}>;
+
+/**
+ * Component providing a recurrent `presence` signal.
+ */
+// Note that "PureComponent" is of questionable veracity: this component's
+// entire purpose is to emit network calls for their observable side effects
+// as a side effect of being rendered.
+//
+// However, it is at least true that there is never a need to call `render()`
+// (and therefore `componentDidUpdate()`) if the props haven't changed.
+class PresenceHeartbeat extends PureComponent<Props> {
+  /** Callback for Heartbeat object. */
+  onHeartbeat = () => {
+    if (this.props.auth) {
+      this.props.dispatch(reportPresence(true));
+    }
+  };
+
+  heartbeat: Heartbeat = new Heartbeat(this.onHeartbeat, 1000 * 60);
+
+  componentDidMount() {
+    this.updateHeartbeatState(); // conditional start
+    AppState.addEventListener('change', this.updateHeartbeatState);
+  }
+
+  componentWillUnmount() {
+    AppState.removeEventListener('change', this.updateHeartbeatState);
```

Comment:
We had a comment thread on this bit over on the previous PR: https://github.com/zulip/zulip-mobile/pull/3704#discussion_r352959826

Your explanation there makes sense to me! I'm convinced.

I also wanted to mention that thread here because you'd said:
> (On the other hand, I'd argue that the lone existing use of `.isActive`, over in `src/events/eventMiddleware.js`, is a bug! Or at least is closely involved in one. But that's not directly relevant here; I'll elaborate somewhere else.)

and I don't want that thought to get lost. :slightly_smiling_face: Want to elaborate on that -- perhaps in a new issue, or a chat thread?

Reply:
Eh, I might as well express my current thoughts on that in the form of commit messages.

Opened PR #3936.